src/iuse.cpp

```diff
@@ -1104,12 +1098,12 @@ int iuse::plantblech(player *p, item *it, bool, const tripoint &pos)
         } else{
             p->add_msg_if_player(m_good, _("Oddly enough, this doesn't taste so bad."));
         }
-        const auto food = dynamic_cast<const it_comest*>(it->type);
+
         //reverses the harmful values of drinking fertilizer
-        p->mod_hunger(p->nutrition_for(food) * multiplier);
-        p->mod_thirst(-food->quench * multiplier);
-        p->mod_healthy_mod(food->healthy * multiplier, food->healthy * multiplier);
-        p->add_morale(MORALE_FOOD_GOOD, -10 * multiplier, 60, 60, 30, false, food);
+        p->mod_hunger( p->nutrition_for( it->type ) * multiplier );
+        p->mod_thirst( -it->type->comestible->quench * multiplier);
```

Comment:
In the `caff` above, you check for comestible type. Here you don't. Fertilizer is more likely to become non-comestible than coffee.